File: app-java/src/main/java/com/zeoflow/memo/demo/MainActivity.java
package com.zeoflow.memo.demo;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.zeoflow.memo.ConcealEncryption;
import com.zeoflow.memo.Memo;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        memoExample();
    }

    private void memoExample() {
        timeMemoInit();
        timeMemoPut();
        timeMemoGet();
        timeMemoContains();
        timeMemoCount();
        timeMemoDelete();
        timeMemoEncrypt();
        timeMemoDecrypt();
    }

    private void timeMemoInit() {
        long startTime = System.currentTimeMillis();

        Memo.Companion.init().withEncryption(
                new ConcealEncryption("dgdffhghdfhfgh")
        ).build();

        long endTime = System.currentTimeMillis();
        System.out.println("Memo.init: " + (endTime - startTime) + "ms");
    }

    private void timeMemoPut() {
        long startTime = System.currentTimeMillis();

        Memo.Companion.put("key", "value");

        long endTime = System.currentTimeMillis();
        System.out.println("Memo.put: " + (endTime - startTime) + "ms");
    }

    private void timeMemoGet() {
        long startTime = System.currentTimeMillis();

        Memo.Companion.get("key");

        long endTime = System.currentTimeMillis();
        System.out.println("Memo.get: " + (endTime - startTime) + "ms");
    }

    private void timeMemoCount() {
        long startTime = System.currentTimeMillis();

        Memo.Companion.count();

        long endTime = System.currentTimeMillis();
        System.out.println("Memo.count: " + (endTime - startTime) + "ms");
    }

    private void timeMemoContains() {
        long startTime = System.currentTimeMillis();

        Memo.Companion.contains("key");

        long endTime = System.currentTimeMillis();
        System.out.println("Memo.count: " + (endTime - startTime) + "ms");
    }

    private void timeMemoDelete() {
        long startTime = System.currentTimeMillis();

        Memo.Companion.delete("key");

        long endTime = System.currentTimeMillis();
        System.out.println("Memo.count: " + (endTime - startTime) + "ms");
    }

    private void timeMemoEncrypt() {
        long startTime = System.currentTimeMillis();

        System.out.println("e: " + Memo.Companion.encrypt(42335));

        long endTime = System.currentTimeMillis();
        System.out.println("Memo.encrypt: " + (endTime - startTime) + "ms");
    }

    private void timeMemoDecrypt() {
        long startTime = System.currentTimeMillis();

        System.out.println("v: " + Memo.Companion.decrypt(Memo.Companion.encrypt(42335)));

        long endTime = System.currentTimeMillis();
        System.out.println("Memo.decrypt: " + (endTime - startTime) + "ms");
    }
}

File: app-kotlin/src/main/java/com/zeoflow/memo/demo/MainActivity.kt
package com.zeoflow.memo.demo

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.zeoflow.memo.ConcealEncryption
import com.zeoflow.memo.Memo
import com.zeoflow.memo.demo.entities.CityMemo

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val cityMemo = CityMemo()
        memoExample()
        cityMemo.nameObservable(this) {
            println("name: $it")
        }
    }

    private fun memoExample() {
        timeMemoInit()
        timeMemoPut()
        timeMemoGet()
        timeMemoContains()
        timeMemoCount()
        timeMemoDelete()
        timeMemoEncrypt()
        timeMemoDecrypt()
    }

    private fun timeMemoInit() {
        val startTime = System.currentTimeMillis()
        Memo.init().withEncryption(
            ConcealEncryption("2343532sdafg54")
        ).build()
        val endTime = System.currentTimeMillis()
        println("Memo.init: " + (endTime - startTime) + "ms")
    }

    private fun timeMemoPut() {
        val startTime = System.currentTimeMillis()
        Memo.put("key", "value")
        val endTime = System.currentTimeMillis()
        println("Memo.put: " + (endTime - startTime) + "ms")
    }

    private fun timeMemoGet() {
        val startTime = System.currentTimeMillis()
        Memo.get<Any>("key")
        val endTime = System.currentTimeMillis()
        println("Memo.get: " + (endTime - startTime) + "ms")
    }

    private fun timeMemoCount() {
        val startTime = System.currentTimeMillis()
        Memo.count()
        val endTime = System.currentTimeMillis()
        println("Memo.count: " + (endTime - startTime) + "ms")
    }

    private fun timeMemoContains() {
        val startTime = System.currentTimeMillis()
        Memo.contains("key")
        val endTime = System.currentTimeMillis()
        println("Memo.count: " + (endTime - startTime) + "ms")
    }

    private fun timeMemoDelete() {
        val startTime = System.currentTimeMillis()
        Memo.delete("key")
        val endTime = System.currentTimeMillis()
        println("Memo.count: " + (endTime - startTime) + "ms")
    }

    private fun timeMemoEncrypt() {
        val startTime = System.currentTimeMillis()
        println("e: " + Memo.encrypt(42335))
        val endTime = System.currentTimeMillis()
        println("Memo.encrypt: " + (endTime - startTime) + "ms")
    }

    private fun timeMemoDecrypt() {
        val startTime = System.currentTimeMillis()
        println("v: " + Memo.decrypt(Memo.encrypt(42335)))
        val endTime = System.currentTimeMillis()
        println("Memo.decrypt: " + (endTime - startTime) + "ms")
    }
}